src/display/annotation_layer.js

```diff
@@ -758,28 +790,38 @@ class PopupAnnotationElement extends AnnotationElement {
     }
 
     const selector = `[data-annotation-id="${this.data.parentId}"]`;
-    const parentElement = this.layer.querySelector(selector);
-    if (!parentElement) {
+    const parentElement = this.layer.querySelectorAll(selector);
+    if (!parentElement || parentElement.length === 0) {
       return this.container;
     }
 
     const popup = new PopupElement({
       container: this.container,
-      trigger: parentElement,
+      trigger: Array.from(parentElement),
       color: this.data.color,
       title: this.data.title,
       modificationDate: this.data.modificationDate,
       contents: this.data.contents,
     });
 
+    // Find the main container if any
+    let main;
```

Comment:
Any reason to not just initialize this here, i.e. `let main = parentElement.item(0);`, since there's no validation below the loop (but rather only a null-check)?